pyro/contrib/funsor/infer/traceenum_elbo.py

```diff
@@ -39,9 +42,72 @@ def terms_from_trace(tr):
         # grab the log-density, found at all sites except those that are not replayed
         if node["is_observed"] or not node.get("replay_skipped", False):
             terms["log_factors"].append(node["funsor"]["log_prob"])
+    # add plate dimensions to the plate_to_step dictionary
+    terms["plate_to_step"].update({plate: terms["plate_to_step"].get(plate, {}) for plate in terms["plate_vars"]})
     return terms
 
 
+@copy_docs_from(_OrigTraceEnum_ELBO)
+class TraceMarkovEnum_ELBO(ELBO):
+
+    def differentiable_loss(self, model, guide, *args, **kwargs):
+
+        # get batched, enumerated, to_funsor-ed traces from the guide and model
+        with plate(size=self.num_particles) if self.num_particles > 1 else contextlib.ExitStack(), \
+                enum(first_available_dim=(-self.max_plate_nesting-1) if self.max_plate_nesting else None):
+            guide_tr = trace(guide).get_trace(*args, **kwargs)
+            model_tr = trace(replay(model, trace=guide_tr)).get_trace(*args, **kwargs)
+
+        # extract from traces all metadata that we will need to compute the elbo
+        guide_terms = terms_from_trace(guide_tr)
+        model_terms = terms_from_trace(model_tr)
+
+        # guide side enumeration is not supported
+        if any(guide_terms["plate_to_step"].values()):
+            raise NotImplementedError("TraceMarkovEnum_ELBO does not yet support guide side enumeration")
```

Comment:
raise `NotImplementedError`

Reply:
nit: change error message to `"TraceMarkovEnum_ELBO does not yet support guide side Markov enumeration"`